query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryServiceImpl.java

```diff
@@ -16,14 +18,40 @@
 import org.hypertrace.core.query.service.api.QueryServiceGrpc;
 import org.hypertrace.core.query.service.api.ResultSetChunk;
 import org.hypertrace.core.query.service.validation.QueryValidator;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
 
 @Singleton
 @Slf4j
 class QueryServiceImpl extends QueryServiceGrpc.QueryServiceImplBase {
+
   private final RequestHandlerSelector handlerSelector;
   private final QueryTransformationPipeline queryTransformationPipeline;
   private final QueryValidator queryValidator;
 
+  private Counter serviceResponseErrorCounter;
+  private Counter serviceResponseSuccessCounter;
+  private static final String ERROR_COUNTER_NAME = "hypertrace.query-service.response.errors";
+  private static final String SUCCESS_COUNTER_NAME = "hypertrace.query-service.response.success";
+
+  class QueryServiceObserver<T> extends ServerCallStreamRxObserver<T> {
```

Comment:
Can you move this to a separate file?

Would it make sense to move `org.hypertrace.core.grpcutils.server.rx` of repo https://github.com/hypertrace/java-grpc-utils/tree/main/grpc-server-rx-utils? Something like as below

`class ServerCallStreamRxObserverWihMetricRecorder<T> extends ServerCallStreamRxObserver<T>` ? 

cc: @aaron-steinfeld

Reply:
hmmmm... Okay, I got your point. Made the change.
Can you check again? Let me know if it's fine.